sdk/test-utils/recorder-new/MIGRATION.md

```diff
@@ -0,0 +1,278 @@
+| `unit-test:browser`        | `karma start --single-run`                                                                                                                                                                    | `dev-tool run test:browser`                                                                                                   |
+| `unit-test:node`           | `mocha -r esm -r ts-node/register --reporter ../../../common/tools/mocha-multi-reporter.js --timeout 1200000 --full-trace --exclude \"test/**/browser/*.spec.ts\" \"test/**/*.spec.ts\"`      | `dev-tool run test:node-ts-input -- --timeout 1200000 --full-trace --exclude 'test/**/browser/*.spec.ts' 'test/**/*.spec.ts'` |
+| `integration-test:browser` | `karma start --single-run`                                                                                                                                                                    | `dev-tool run test:browser`                                                                                                   |
+| `integration-test:node`    | `nyc mocha -r esm --require source-map-support/register --reporter ../../../common/tools/mocha-multi-reporter.js --timeout 5000000 --full-trace \"dist-esm/test/{,!(browser)/**/}*.spec.js\"` | `dev-tool run test:node-js-input -- --timeout 5000000 'dist-esm/test/**/*.spec.js'`                                           |
+
+Note the difference between the dev-tool `node-ts-input` and `node-js-input` commands:
+
+- `node-ts-input` runs the tests using `ts-node`, without code coverage.
+- `node-js-input` runs the tests using the built JavaScript output, and generates coverage reporting using `nyc`.
+
+## Initializing the recorder
+
+The approach taken to initialize the recorder depends on whether the SDK being tested uses Core v1 ([`core-http`]) or Core v2 ([`core-rest-pipeline`]). If your SDK is on Core v2, read on. If you're still on Core v1, [jump to the section on Core v1 below](#for-core-v1-sdks).
+
+### For Core v2 SDKs
+
+The recorder is implemented as a custom policy which should be attached to your client's pipeline. Firstly, initialize the recorder:
+
+```ts
+let recorder: Recorder;
+
+/*
+ * Note the use of function() instead of the arrow syntax. We need access to `this` so we
+ * can pass test information from Mocha to the recorder.
+ */
+beforeEach(function (this: Context) {
+  recorder = new Recorder(this.currentTest);
+});
+```
+
+To enable the recorder, you should then initialize your SDK client as normal and use the recorder's `configureClient` method. This method will attach the necessary policies to the client for recording to be enabled.
```

Comment:
Let's also mention that the client should be exposing the "pipeline" object to get the policy added properly.

Reply:
It would be a compile error if `pipeline` isn't exposed.